client/header/activity-panel/panels/inbox/index.js

```diff
@@ -54,7 +48,7 @@ const renderNotes = ( {
 	const notesArray = Object.keys( notes ).map( ( key ) => notes[ key ] );
 
 	return notesArray.map( ( note ) => {
-		if ( isDismissUndoRequesting === note.id ) {
+		if ( note.isUpdating ) {
```

Comment:
When we are updating a note (when undoing the dismission) `note.isUpdating` is `undefined` but should be `true` to be able to show the placeholder in its place.